### ISSUE TITLE
Cannot free memory from loaded model unless a builder is created

### ISSUE THREAD

**Issue by Lyrcaxis** [sandrohanea/whisper.net#256]
I found out that calling `whisperFactory.Dispose()` does not free the memory taken by the loaded whisper model unless the `contextLazy.Value` is accessed (from Builder).

seems contextLazy is a little too lazy in this case:
![image](https://github.com/user-attachments/assets/c315ba1c-f3cb-4adc-9fcc-9f9348c1d222)

Simply switching to `if (contextLazy.Value != IntPtr.Zero)` does the trick but I'm not sure whether it defeats the purpose of lazy init.

minimal repro:
```
void Test() { using var whisperFactory = WhisperFactory.FromPath("model.bin"); }
```

**Comment by sandrohanea** [comment]
Hello @Lyrcaxis ,
Indeed, it seems there is a problem as we also use "eager initialization". The Lazy<> is used only if `delayInitialization` is explicit when the factory is created.

Preparing a fix now.

**Comment by sandrohanea** [comment]
Fix was merged and released as 1.7.2
